core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -326,7 +326,7 @@ public void onLoad(Run<?, ?> r) {
         for (ParameterValue v : this.parameters) {
             if (this.parameterDefinitionNames.contains(v.getName()) || isSafeParameter(v.getName())) {
                 filteredParameters.add(v);
-            } else if ("false".equalsIgnoreCase(shouldKeepFlag)) {
+            } else if (!"false".equalsIgnoreCase(shouldKeepFlag)) {
```

Comment:
Generally this string comparison is not correct (`foo` string will be `false` as well), `SystemProperties.getBoolean()` should be used instead to cover all `false` strings. Same for the comparison above.